extensions/libraries/redcore/form/fields/rdatepicker.php

```diff
@@ -49,6 +49,109 @@ class JFormFieldRdatepicker extends JFormFieldRtext
 	public $datepickerOptions = '';
 
 	/**
+	 * The filter.
+	 *
+	 * @var    integer
+	 */
+	protected $filter;
+
+	protected $symbolsMatching = array(
+		// Day
+		'dd' => 'd',
+		'D' => 'D',
+		'd' => 'j',
+		'DD' => 'l',
+		'o' => 'z',
+		// Month
+		'MM' => 'F',
+		'mm' => 'm',
+		'M' => 'M',
+		'm' => 'n',
+		// Year
+		'yy' => 'Y',
+		'y' => 'y',
+		'oo' => 'z',
+		'@' => 'U',
+		'!' => 'u',
+		'ATOM' => JDate::ATOM,
+		'COOKIE' => JDate::COOKIE,
+		'ISO_8601' => JDate::ISO8601,
+		'RFC_822' => JDate::RFC822,
+		'RFC_850' => JDate::RFC850,
+		'RFC_1036' => JDate::RFC1036,
+		'RFC_1123' => JDate::RFC1123,
+		'RFC_2822' => JDate::RFC2822,
+		'RSS' => JDate::RSS,
+		'TICKS' => '\!',
+		'TIMESTAMP' => 'U',
+		'W3C' => JDate::W3C,
+	);
+
+	/**
+	 * Method to attach a JForm object to the field.
+	 *
+	 * @param   SimpleXMLElement  $element  The SimpleXMLElement object representing the `<field>` tag for the form field object.
+	 * @param   mixed             $value    The form field value to validate.
+	 * @param   string            $group    The field name group control value. This acts as as an array container for the field.
+	 *                                      For example if the field has name="foo" and the group value is set to "bar" then the
+	 *                                      full field name would end up being "bar[foo]".
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @see     JFormField::setup()
+	 */
+	public function setup(SimpleXMLElement $element, $value, $group = null)
+	{
+		$return = parent::setup($element, $value, $group);
+
+		if ($return)
+		{
+			$this->filter = (string) $this->element['filter'] ? (string) $this->element['filter'] : '';
+		}
+
+		return $return;
+	}
+
+	/**
+	 * Method to get certain otherwise inaccessible properties from the form field object.
+	 *
+	 * @param   string  $name  The property name for which to the the value.
+	 *
+	 * @return  mixed  The property value or null.
+	 */
+	public function __get($name)
```

Comment:
Because there is a standard joomla approach. I just expanded already existed magic method in the joomla class.